lib/matplotlib/pyplot.py

```diff
@@ -797,7 +867,8 @@ def figure(num=None,  # autoincrement if None, else integer from 1-N
     else:
         num = int(num)  # crude validation of num argument
 
-    manager = _pylab_helpers.Gcf.get_fig_manager(num)
+    # Type of "num" has narrowed to int, but mypy can't quite see it
+    manager = _pylab_helpers.Gcf.get_fig_manager(num)  # type: ignore[arg-type]
```

Comment:
This falls into the category of things where the implementation certainly allows it to some degree, but I'm curious about the reasons why it is actually allowed/whether it is actually encouraged... it is referred to as "number" everywhere except for one parenthetical in a docstring...

And if we look deeper than "these are the keys of an internal dictionary", it gets murkier:

The fact that just above this we use `int(num)` as a "crude validation" if num is not a str/None makes me think that it actually _doesn't_ fully work with non integer values, at least not as expected, for figures generated through `plt.figure` and if that's the case, I think the course of action is to remove the parenthetical from the docstring... (the Gcf class in and of itself, certainly does work with any hashable, but it is a class defined in a private module and only really used for the purposes of `pyplot`, so if it doesn't work in that context, I'm not sure why it should be documented to work at all)

For that matter, the implementation of `plt.figure` would lead me to believe it only works properly if the labels are specifically `str` not `Any`...